### ISSUE TITLE
[FR] distinguish Do53, DoT and DoH in the logs

### ISSUE THREAD

**Issue by andreasschulze** [NLnetLabs/unbound#969]
the Problem is discussed in the list archive: https://lists.nlnetlabs.nl/pipermail/unbound-users/2023-November/008188.html

Yorgos suggested to open a FR ticket here. I like his idea to simply log the destination port and 
suggest to _append_ that information to the current log. for example here: https://github.com/NLnetLabs/unbound/blob/15a9b0f2be1b5d9d53806c817ed52bc88f8f3a60/util/data/msgreply.c#L919

**Comment by andreasschulze** [comment]
while "my" use-case would be solved by logging a destination _port_, other users may see benefits if the destination _address_ is logged. It may make sense to log _both_ if an new option `log-destaddr: yes` is set (or split in `log-destaddr` and `log-destport`)

**Comment by renaudallard** [comment]
It would also be useful to log the protocol (TCP/UDP).
In the end, we have 53 udp or tcp. 853 tcp for DoT and 853 udp for DoQ. 443 tcp for DoH.

**Comment by pemensik** [comment]
I have found by mistake unbound supports also DTLS on port 853, not necessary reserved for DoQ. Socket stream/datagram indication should be present, but I think indication of protocol used to receive query would be the best. You can still configure to accept DoT on port 443, at least dnssec-trigger uses that.

**Comment by wcawijngaards** [comment]
The option `log-destaddr: yes` can be used print the destination address, port and type, for the log-replies output. It is implemented in the commit to the repository.

The fix adds this type of output to the log replies: `127.0.0.1 localhost. A IN NOERROR 0.000000 1 54 on tcp 127.0.0.1 53`.

**Comment by andreasschulze** [comment]
Hello Wouter, looks like the commit didn't found a way into release 1.19.1. Was this intentional?

**Comment by wcawijngaards** [comment]
The 1.19.1 release is a security release, and contains the security fixes only. The code is there waiting for a next release, in the code repository.

**Comment by sunbearc22** [comment]
Is the `log-destaddr:` attribute supposed to be available in 1.19.2? I can't use it from the Unbound package provided in Ubuntu Server 24.04.1
 
```
root@DNS:/etc/unbound# unbound-checkconf unbound.conf
/etc/unbound/unbound.conf:32: error: unknown keyword 'log-destaddr'
/etc/unbound/unbound.conf:32: error: stray ':'
/etc/unbound/unbound.conf:32: error: stray '"'
/etc/unbound/unbound.conf:32: error: unknown keyword 'yes'
/etc/unbound/unbound.conf:32: error: stray '"'
read unbound.conf failed: 5 errors in configuration file
root@DNS:/etc/unbound# unbound -V
Version 1.19.2

Configure line: --build=aarch64-linux-gnu --prefix=/usr --includedir=${prefix}/include --mandir=${prefix}/share/man --infodir=${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --disable-option-checking --disable-silent-rules --libdir=${prefix}/lib/aarch64-linux-gnu --runstatedir=/run --disable-maintainer-mode --disable-dependency-tracking --with-pythonmodule --with-pyunbound --enable-subnet --enable-dnstap --enable-systemd --enable-cachedb --with-libhiredis --with-libnghttp2 --with-chroot-dir= --with-dnstap-socket-path=/run/dnstap.sock --disable-rpath --with-pidfile=/run/unbound.pid --with-libevent --enable-tfo-client --with-rootkey-file=/usr/share/dns/root.key --disable-flto --enable-tfo-server
Linked libs: libevent 2.1.12-stable (it uses epoll), OpenSSL 3.0.13 30 Jan 2024
Linked modules: dns64 python cachedb subnetcache respip validator iterator
TCP Fastopen feature available

BSD licensed, see LICENSE in source package for details.
Report bugs to unbound-bugs@nlnetlabs.nl or https://github.com/NLnetLabs/unbound/issues
```

**Comment by gthess** [comment]
[1.19.2 was also a security release](https://nlnetlabs.nl/projects/unbound/download/#unbound-1-19-2). The option made its way in the 1.19.3 release.
